src/utils/pathCheck.ts

```diff
@@ -0,0 +1,112 @@
+import fs from 'fs';
+import path from 'path';
+
+import semver from 'semver';
+import * as childProcess from '@/utils/childProcess';
+import resources from '@/resources';
+
+const flags: Record<string, string> = {
+  helm:    'version',
+  kim:     '-v',
+  kubectl: 'version',
+};
+const regexes: Record<string, RegExp> = {
+  // helm has to match both
+  // current: version.BuildInfo{Version:"v3.5.3", ...
+  // older:   Client: &version.Version{SemVer:"v2.16.12", ...
+  helm:    /Version.*:.*?"v(.+?)"/,
+  kim:     /version v(\S+)/,
+  kubectl: /Client Version.*?GitVersion:"v(.+?)"/,
+};
+
+export default async function shadowInfo(targetDir: string, binaryName: string): Promise<Array<string>> {
+  const notes: Array<string> = [];
+  const referencePath = resources.executable(binaryName);
+
+  try {
+    await fs.promises.access(referencePath, fs.constants.R_OK | fs.constants.X_OK);
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      console.log(err);
+
+      return notes;
+    }
+  }
+  const proposedVersion = await getVersion(referencePath, binaryName);
+
+  if (!proposedVersion) {
+    return notes;
+  }
+  const paths: Array<string> = process.env.PATH?.split(path.delimiter) ?? [];
+
+  let sawCurrentDir = false;
+
+  for (const currentDir of paths) {
+    if (currentDir === targetDir) {
```

Comment:
Do need to normalize things? The user might have `/usr/local/bin/` (instead of `/usr/local/bin`, note trailing slash) in `$PATH`.

Reply:
Good catch.  `$PATH` is user-defined data, so we need to allow for irregularities